features/commands.js

```diff
@@ -85,6 +85,46 @@ https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html`,
         });
       }
     },
+    {
+      words: [`!stateUpdates`],
+      help: `Explains the issue with state updates being asynchronous`,
+      handleMessage: msg => {
+        msg.channel.send({
+          embed: {
+            title:
+              "State Updates May Be Asynchronous",
+            type: "rich",
+            description: `Often times you might run into an issue like this
+\`\`\`js
+const handleEvent = e => {
+  setState(e.target.value);
+  console.log(state);
+}
+\`\`\`
+where \`state\` is not the most up to date value when you log it. This is caused by state updates being asynchronous, so synchronous logic after a state update should not rely on the state value.
+\`\`\`js
+// log it directly inside render
+console.log(state);
+
+// Or in a useEffect call
+useEffect(() => {
+  console.log(state);
+}, [state]);
+
+const handleEvent = e => {
+  setState(e.target.value);
+};
+\`\`\`
```

Comment:
I feel like this misses the other solution of 
```
const handleEvent = e => {
  const {value} = e.target
  setState(value);
  console.log(value);
};
```
thoughts? It also feels a little long already, maybe drop the fixed code example and reduce how many links it includes? Defining the problem, then offering 3 solutions and 4 links to more resources seems like a lot.

Reply:
Agree on this one:
>Defining the problem, then offering 3 solutions and 4 links to more resources seems like a lot.

Reply:
Okay, omitting the solution from the example makes sense, especially since my gist already covers all three solutions (and has a basic solution up front). Which links should I omit from it though? They all seem pretty valuable to include (I guess I *could* update the gist to link to Dan's resources under a new section, and then just include the original two links)

Reply:
Yeah using the gist and a `further reads on this topic (or sth..)` is probably the smartest idea